BunnyManager.cpp

```diff
@@ -0,0 +1,48 @@
+//
+// Created by Estevan Hernandez on 8/22/17.
+//
+
+#include "BunnyManager.hpp"
+
+BunnyManager* BunnyManager::s_instance = nullptr;
+
+void BunnyManager::addBunny(Bunny *bunny) {
+    static bool debug;
+    bool shouldAdd = true;
+    for (Bunny *aBunny: bunnies_) {
+        if (aBunny == bunny) {
+            shouldAdd = false;
+            break;
+        }
+    }
+    if (shouldAdd == true) {
+        this->bunnies_.push_back(bunny);
+    } else {
+        if (debug) std::cout << "That Bunny is already in the Bunny Manager." << std::endl;
+    }
+
+}
+
+void BunnyManager::removeBunny(Bunny *bunny) {
+    bunnies_.erase(std::remove(bunnies_.begin(), bunnies_.end(), bunny), bunnies_.end());
+}
+
+bool BunnyManager::isABunnyTeleporting() {
+    for (Bunny *bun: bunnies_) {
```

Comment:
this could be changed later to make it so that there is a bool that a bunny might change if it's state is teleporting, and would change back when it's done. that way its O(1) instead of O(n). worst case scenario youre iterating through all bunnies every frame to check if any are teleporting

Reply:
Oh I like that!